### PR TITLE
Edit dev install docs and add troubleshooting tips

### LINKED ISSUE #732: Add a Troubleshooting subsection to dev install docs
Based on discussion on gitter and #662, let's expand the notes subsection into a Troubleshooting subsection to https://github.com/ipython/ipywidgets/blob/master/docs/source/dev_install.md 

While it might not solve the underlying issue, it will at least provide greater possiblity for users to find a solution when install is 😢 

cc @fperez @SylvainCorlay

### PR DESCRIPTION
**Pull request by willingc**
Closes #732 

@SylvainCorlay @fperez

### CODE REVIEW

File: docs/source/dev_install.md [hunk 1]
```diff
@@ -57,13 +59,13 @@ permissions on npm and pip related install directories are correct.
 - If troubleshooting an upgrade and its build, you may need to do the
   following process:
   
-      - Deep clean of the cloned repository:
+    - Deep clean of the cloned repository:
       
-             git clean -dfx .
+            git clean -dfx .
```
Comment (fperez) [inline docs/source/dev_install.md]: shouldn't this be in a triple-backticks block so it renders as code?

File: docs/source/dev_install.md [hunk 2]
```diff
@@ -57,13 +59,13 @@ permissions on npm and pip related install directories are correct.
...
 
-      - Remove anything with `widgetsnbextension` in the name of files within
+    - Remove anything with `widgetsnbextension` in the name of files within
         the `conda` directory
         
-      - Remove `node_modules` directory
+    - Remove `node_modules` directory
```
Comment (fperez) [inline docs/source/dev_install.md]: Actually, is this necessary as a separate step? I'd have thought the `git clean ...` would have nuked it already...

Comment (willingc) [inline docs/source/dev_install.md]: @fperez It may not be. Wasn't sure if `git clean` would nuke it. @SylvainCorlay ?

Comment (willingc) [inline docs/source/dev_install.md]: Nevermind. Just tested it out. @fperez you were absolutely correct :smile:

**Comment by fperez** [comment]
thanks @willingc! BTW, I just realized the triple-backticks weren't really needed b/c you'd already indented the source, which makes markdown render it as code. Sorry about that :)

Let's see if @SylvainCorlay has a chance to give it a once-over, since he's much more familiar with the process than me.

Otherwise I'm +1 on merging this, thanks!!

**Comment by jdfreder** [comment]
Fantastic!  Thanks @willingc

**Comment by SylvainCorlay** [comment]
👍

**Pull request merged**